### ISSUE TITLE
Race between CREATE MAPPING and using that mapping

### ISSUE THREAD

**Issue by viliam-durina** [hazelcast/hazelcast-jet#2708]
The `SqlClientTest.test_jetJobReturnRowsToClientFrom` test failed with:
```
com.hazelcast.sql.HazelcastSqlException: At line 1, column 15: Object 't' not found
	at com.hazelcast.sql.impl.client.SqlClientService.handleResponseError(SqlClientService.java:220)
	at com.hazelcast.sql.impl.client.SqlClientService.execute(SqlClientService.java:100)
	at com.hazelcast.sql.SqlService.execute(SqlService.java:144)
	at com.hazelcast.jet.sql.SqlClientTest.test_jetJobReturnRowsToClientFrom(SqlClientTest.java:55)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:59)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:56)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at com.hazelcast.test.FailOnTimeoutStatement$CallableStatement.call(FailOnTimeoutStatement.java:114)
	at com.hazelcast.test.FailOnTimeoutStatement$CallableStatement.call(FailOnTimeoutStatement.java:106)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.lang.Thread.run(Thread.java:748)
```

http://jenkins.hazelcast.com/job/jet-oss-pr-builder/3676/testReport/junit/com.hazelcast.jet.sql/SqlClientTest/test_jetJobReturnRowsToClientFrom/

I suspect the cause is that the underlying ReplicatedMap wasn't yet replicated to the other member.

**Comment by olukas** [comment]
Failed again on Oracle JDK 8: http://jenkins.hazelcast.com/job/jet-oss-master-oracle-jdk8/315/testReport/junit/com.hazelcast.jet.sql/SqlClientTest/test_jetJobReturnRowsToClientFrom/

**Comment by viliam-durina** [comment]
again: http://jenkins.hazelcast.com/job/jet-oss-pr-builder/3858/testReport/junit/com.hazelcast.jet.sql/SqlClientTest/test_jetJobReturnRowsToClientFrom/

**Comment by ufukyilmaz** [comment]
Failed again on pr-builder: http://jenkins.hazelcast.com/job/jet-oss-pr-builder/3906/testReport/junit/com.hazelcast.jet.sql/SqlClientTest/test_jetJobReturnRowsToClientFrom/

**Comment by viliam-durina** [comment]
Mostly fixed by #2790
